server.go

```diff
@@ -31,10 +32,11 @@ var (
 
 // Server is a mailing list server.
 type Server struct {
-	cfg Config
-	db  *bolt.DB
-	sck net.Listener
-	msg chan *Message
+	cfg       Config
+	db        *bolt.DB
+	sck       net.Listener
+	msg       chan *Message
+	boltStore *database.BoltDataStore
```

Comment:
using the explicit type here kind of defeats the purpose of introducing an interface :)

we should drop the `db *bolt.DB` field and replace `boltStore` with:

```go
type Server struct {
    // ...
    db database.Store
}
```

assuming the interface `Store` is defined in the `database` package:

```go
package database

type Store interface {
     Subscribers(list string) ([]string, error)
￼     Subscribe(user, list string) error
     Usubscribe(user, list string) error
     Lists() ([]string, error)
     Users() ([]string, error)
}
```

then, the config should be modified to specify the database name/type to use.
name to be used later on with your `database.New()` or, rather: `database.Open("boltdb", cfg.Database)`

Reply:
Agreed with that. Conrete type is to be replaced by interface but BoltDBStore didn't implement all the methods of interface so it was remaining. I'll do that now.